test/cpp/api/modules.cpp

```diff
@@ -1009,6 +1009,32 @@ TEST_F(ModulesTest, CosineEmbeddingLoss) {
   ASSERT_EQ(input2.sizes(), input2.grad().sizes());
 }
 
+TEST_F(ModulesTest, SmoothL1LossDefaultOptions) {
+  SmoothL1Loss loss;
+  auto input = torch::tensor({{0.1}, {1.2}, {4.7}}, torch::requires_grad());
+  auto target = torch::tensor({{0.}, {1.}, {5.}}, torch::kFloat);
+  auto output = loss(input, target);
+  auto expected = torch::tensor({0.0233335}, torch::kFloat);
+  auto s = output.sum();
+  s.backward();
+
+  ASSERT_TRUE(output.allclose(expected));
+  ASSERT_EQ(input.sizes(), input.grad().sizes());
+}
+
+TEST_F(ModulesTest, SmoothL1LossNoReduction) {
+  SmoothL1Loss loss(SmoothL1LossOptions().reduction(Reduction::None));
```

Comment:
We could write this as
```suggestion
  SmoothL1Loss loss(/*reduction=*/Reduction::None);
```
to also test the implicit conversion for `SmoothL1LossOptions` :D

Reply:
Hi @yf225! Taking your previous reviews into account, I wrote these 4 tests for each _single-optional-parameter-loss_ module/functional pair, covering explicit instantiation (in `modules.cpp`, `*DefaultOptions`), explicit instantiation + changing parameter value (in `modules.cpp`, `*NoReduction`, this test case), omitting optional parameter for functional (in `functional.cpp`, `*DefaultOptions`) and implicit instantiation as a parameter in functional (in `functional.cpp`, `*NoReduction`).

Do you see any other test that should be run for these modules/functionals and their options? (I ask this here because you already corrected the others hehe)